pkgs/racket-doc/scribblings/reference/equality.scrbl

```diff
@@ -634,6 +655,42 @@ and only access mutable data when the mode is true:
 ]
 ]
 
+Certain type-specific equality functions on mutable types,
+such as @racket[string=?], should also be avoided unless the
+struct is mutable or the mode is true.
+Type-specific equality functions on immutable types, such as
+@racket[symbol=?], are fine.
+
+@compare0[#:left "fine" #:right "bad"
+@racketblock0[
+  (struct thing (name)
+    (code:comment "symbols are immutable: no problem")
+    #:methods gen:equal+hash
+    [(define (equal-proc self other rec)
+       (symbol=? (thing-name self) (thing-name other)))
+     (define (hash-proc self rec)
+       (+ (eq-hash-code struct:thing)
```

Comment:
Should this use a hash code mixing function?

Reply:
Mm. Maybe, but it doesn't feel like that's the relevant thing here. Maybe the example would be clearer without a `hash-proc` at all. Just the definition of `equal-proc` is enough, as long as it's clear that the struct it's for is immutable.